src/riak_core_apl.erl

```diff
@@ -40,9 +43,12 @@
 -type ring() :: riak_core_ring:riak_core_ring().
 -type preflist() :: [{index(), node()}].
 -type preflist_ann() :: [{{index(), node()}, primary|fallback}].
+-type preflist_with_number_ann() :: [{{partition_id(), node()}, primary|fallback}].
 -type iterator() :: term().
 -type chashbin() :: term().
 -type docidx() :: chash:index().
+-type partition_id() :: non_neg_integer().
+-type ring_size() :: non_neg_integer().
```

Comment:
We could probably define this in `riak_core_ring` and normalize the fact that in that file we use both `integer` and `pos_integer` for it, so everything uses the same type.